l10n_ch_account_statement_base_import/parsers/raiffeisen_file_parser.py

```diff
@@ -0,0 +1,285 @@
+
+    def _parse_transactions(self):
+        """Parse bank statement lines from file
+        list of dict containing :
+            - 'name': string (e.g: 'KBC-INVESTERINGSKREDIET 787-5562831-01')
+            - 'date': date
+            - 'amount': float
+            - 'unique_import_id': string
+            -o 'account_number': string
+                Will be used to find/create the res.partner.bank in odoo
+            -o 'note': string
+            -o 'partner_name': string
+            -o 'ref': string
+
+        :return: a list of transactions
+        :rtype: list
+        """
+
+        transactions = []
+        id = 0
+        for row in self.rows:
+            transactions.append(self.get_st_line_vals(row, id))
+            id += 1
+
+        return transactions
+
+    def _get_values(self, line, currency='CHF', rate=1):
+        name = ''
+        amount = '0.0'
+        vals = line.get('Text').rsplit(' ' + currency + ' ', 1)
+        if len(vals) == 2:
+            name = vals[0]
+            amount = float(vals[1].replace("'", "")) * rate
+        else:
+            raise exceptions.Warning(
+                'ParsingError', _('Unable to parse amount for ligne %s') %
+                line.get('Text'))
+        return name, str(amount)
+
+    def get_st_line_vals(self, line, id):
+        """
+        This method must return a dict of vals that can be passed to create
+        method of statement line in order to record it.
+            :param:  line: a dict of vals that represent a line of
+                           result_row_list
+            :return: dict of values to give to the create method of
+                     statement line,
+        """
+
+        # We try to extract a BVR reference
+        result = re.match(r'.*(\d{27}).*', line.get('Text'))
+        ref = '/'
+        test = 'Cr�dit'.decode('iso-8859-15').encode('utf8')
+        if result and not line.get('Text').startswith(test):
+            ref = result.group(1)
+
+        date = fields.Date.from_string(line.get("Booked At",
+                                                datetime.date.today()))
+        res = {
+            'name': line.get("Text"),
+            'date': line.get("Booked At", date),
+            'amount': line.get("Credit/Debit Amount", 0.0),
+            'ref': ref,
+            'unique_import_id': str(uuid.uuid4())
+        }
+
+        return res
+
+    def cleanup_rows(self, rows):
+        """
+        Clean up rows to remove useless details for the statement line
+        """
+
+        cleanup_rows = []
+        last_date = ''
+        reported_text = ''
+        reported_line = {}
+        rate = 1
+        currency = 'CHF'
+        for row in rows:
+            if row.get('Booked At'):  # Main row
+                rate = 1
+                currency = 'CHF'
+                reported_text = ''
+                reported_line = {}
+                last_date = row.get('Booked At')
+                if row.get('Text').startswith('Cr�dit'):

```

Comment:
In fact, both exist in the .csv
![raiff](https://cloud.githubusercontent.com/assets/8435180/9907557/0993b8b8-5c90-11e5-896f-9aabf16d0822.PNG)